.pre-commit-config.yaml

```diff
@@ -0,0 +1,6 @@
+# https://pre-commit.com/
```

Comment:
1. Install `pre-commit`, it's packaged for many repositories, https://repology.org/project/python:pre-commit/versions
2. cd to local clone of this repo
3. `pre-commit install`
4. Whenever running `git commit`, doctoc gets executed

I wonder if I should add a word to contributing.md about it?

Including this file also opens the possibility of using https://pre-commit.ci/

Reply:
Why is that needed?

Reply:
It seems to me to be the easiest way to run doctoc regardless of platform and ensure it's run if headings change.

Reply:
e.g. restoring the deploy section without running doctoc

```
└┌(%:~/src/github.com/Mikaela/awesome-sailfishos)┌- git commit -m "README.md: restore Deploy section"
doctoc...................................................................Failed
- hook id: doctoc
- files were modified by this hook

DocToccing single file "README.md" for github.com.

==================

"README.md" will be updated

Everything is OK.
```

Reply:
> I wonder if I should add a word to contributing.md about it

Yes I think it might be worth making it known that there is a pre-commit configuration